### PR TITLE
Feature/199  file range and conditional request support

### PR DESCRIPTION
**Pull request by inkhey**
related to #199 
note: this include also merge with master to have a less deprecated version number in develop

Add simple support for conditional response (range, if modified, etag, etc…) for pyramid.
Based on existing method:
https://docs.pylonsproject.org/projects/webob/en/stable/api/response.html#webob.response.Response.conditional_response_app

### CODE REVIEW

File: hapic/ext/pyramid/context.py [hunk 1]
```diff
@@ -106,20 +106,24 @@ def get_file_response(self, file_response: HapicFile, http_code: int):
                 # is automatically guessed
                 content_type=file_response.mimetype or None,
             )
+            response.conditional_response = file_response.accept_byte_range
         else:
             from pyramid.response import FileIter
             from pyramid.response import Response
 
-            response = Response(status=http_code)
+            response = Response(status=http_code, conditional_response=file_response.accept_byte_range)
```
Comment (inkhey) [inline hapic/ext/pyramid/context.py:114]: not sure about the condition we should use to active conditional response, maybe we should have 2 field:
- one for accept range header only
- one for conditional response mecanism activation 
?

Comment (inkhey) [reply hapic/ext/pyramid/context.py:114]: i change a bit the behavior in order to support that:
- conditional response is active by default
- other framework don't support this feature now and should not try to return that their does accept bytes.

File: hapic/ext/pyramid/context.py [hunk 2]
```diff
@@ -114,13 +114,24 @@ def get_file_response(self, file_response: HapicFile, http_code: int):
             response.content_type = file_response.mimetype
             response.app_iter = FileIter(file_response.file_object)
 
+        response.conditional_response = file_response.use_conditional_response
```
Comment (grignards) [inline hapic/ext/pyramid/context.py:117]: Shouldn't the `file_response` parameter be `hapic_file/file` instead ? The "real" `FileResponse` object is named `response` which is confusing. (I know this is not coming from your PR so feel free to ignore this comment).

**Comment by grignards** [comment]
Also we could have an `NotImplementedError` exception raised in the other framework extensions if the `use_conditional_response` is set to `True`.

**Comment by inkhey** [comment]
> Also we could have an `NotImplementedError` exception raised in the other framework extensions if the `use_conditional_response` is set to `True`.

If we do this, we need to change some other things, as use_conditional_response is set to true by default, if we do not do this, this mean refactoring in each usage of HapicFile which is not really practical as this is most probably nice to have in most case.

File: hapic/ext/pyramid/context.py (hunk 2 above)
Comment (inkhey) [reply hapic/ext/pyramid/context.py:117]: That's a good suggestion, but i clearly prefer to to change such things in this PR.

**Comment by grignards** [comment]
> If we do this, we need to change some other things, as use_conditional_response is set to true by default, if we do not do this, this mean refactoring in each usage of HapicFile which is not really practical as this is most probably nice to have in most case.

I didn't notice the default value for `use_conditional_response`. This makes this PR a behavior change for all hapic users (with pyramid). But OK since the goal here is to go fast.

**Pull request merged**